src/main/kotlin/lotto/domain/vo/LottoNumber.kt

```diff
@@ -0,0 +1,21 @@
+package lotto.domain.vo
+
+@JvmInline
+value class LottoNumber(val number: Int) : Comparable<LottoNumber> {
+    init {
+        require(number in MIN_VALUE..MAX_VALUE) { "lotto number should be between $MIN_VALUE and $MAX_VALUE" }
+    }
+
+    companion object {
+        private const val MIN_VALUE = 1
+        private const val MAX_VALUE = 45
+    }
+
+    override fun compareTo(other: LottoNumber): Int {
+        return when {
+            this.number > other.number -> 1
+            this.number < other.number -> -1
+            else -> 0
+        }
```

Comment:
```suggestion
        return number.compareTo(other.number)
        return compareValues(number, other.number
```
직접 구현할 수도 있지만 코틀린에서 제공하는 기본적인 타입들은 안전하게 코틀린에서 제공하는 함수들로 비교가 가능합니다.